### PR TITLE
[Feature] add dockerfile

### PR DESCRIPTION
**Pull request by liunaijie**
### Purpose of this pull request
add ci to auto build full connector docker image and publish it into dockerhub

### Check list

* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
* [ ] Update the [`release-note`](https://github.com/apache/seatunnel/blob/dev/release-note.md).

### CODE REVIEW

**Comment by liunaijie** [comment]
the success ci link: https://github.com/liunaijie/seatunnel/actions/runs/10299226521/job/28506078267
dockerhub image link: https://hub.docker.com/r/liunaijie/seatunnel/tags


And we need add some variable into repo setting
image

File: .github/workflows/publish-docker.yaml [hunk 1]
```diff
@@ -36,11 +37,24 @@ jobs:
...
+      - name: Maximize runner space
+        uses: ./.github/actions/maximize-build-space
+        with:
+          root-reserve-mb: 30720
+          temp-reserve-mb: 10240
+          remove-dotnet: 'true'
+          remove-android: 'true'
+          remove-haskell: 'true'
+          remove-codeql: 'true'
+          remove-docker-images: 'true'
```
Comment (Hisoka-X) [inline .github/workflows/publish-docker.yaml]: Please use our own script instead. https://github.com/apache/seatunnel/blob/dev/.github/workflows/backend.yml#L342

Comment (liunaijie) [reply .github/workflows/publish-docker.yaml]: Done

**Review by Hisoka-X — APPROVED** [review]

**Comment by Hisoka-X** [comment]
Thanks @liunaijie ！

**Review by hailin0 — APPROVED** [review]

File: pom.xml [hunk 2]
```diff
@@ -760,6 +764,95 @@
...
+                                <environmentVariables>
+                                    <DOCKER_BUILDKIT>1</DOCKER_BUILDKIT>
+                                </environmentVariables>
+                                <executable>docker</executable>
+                                <workingDirectory>${project.basedir}</workingDirectory>
+                                <arguments>
+                                    <argument>buildx</argument>
+                                    <argument>build</argument>
+                                    <argument>--platform</argument>
+                                    <argument>linux/amd64,linux/arm64</argument>
```
Comment (Hisoka-X) [inline pom.xml:842]: Our image supported arm64 now?

File: docs/en/start-v2/docker/docker.md [hunk 3]
```diff
@@ -3,7 +3,104 @@
...
+```
+
+How to submit job in local mode
+
+```shell
+docker run --rm -it apache/seatunnel bash ./bin/seatunnel.sh -e local -c <CONFIG_FILE>
+
+
+# eg: a fake source to console sink
+docker run --rm -it apache/seatunnel bash ./bin/seatunnel.sh -e local -c config/v2.batch.config.template
```
Comment (Hisoka-X) [inline docs/en/start-v2/docker/docker.md:23]: Can we use config file not in docker image?

**Comment by Hisoka-X** [comment]
Waiting infra add docker token. https://issues.apache.org/jira/browse/INFRA-26068

**Pull request merged**